src/components/calcite-pick-list/shared-list-logic.ts

```diff
@@ -123,7 +123,25 @@ export const sharedListMethods = {
     const filteredData = event.detail;
     const values = filteredData.map((item) => item.value);
     this.items.forEach((item) => {
+      const inGroup = item.parentElement.matches("calcite-pick-list-group");
+
       item.hidden = values.indexOf(item.value) === -1;
+
+      // If item is in a group...
+      if (inGroup) {
+        const groupParent = item.parentElement.querySelector("[slot=parent-item]") as pickOrValueListItem;
+        // If there is a group parent
+        if (groupParent !== null) {
+          // If the group parent is a match, show me.
+          if (values.indexOf(groupParent.value) !== -1) {
```

Comment:
Instead of commenting, can you store the conditions in a self-documenting variable? Also, maybe using `if(values.find(groupParent.value))` could help improve readability and even eliminate the need for the comment.